File: src/python/finished/keep_multiplying_found_values_by_two.py
from typing import List

# https://leetcode.com/problems/keep-multiplying-found-values-by-two/


class Solution:
    def findFinalValue(self, nums: List[int], original: int) -> int:
        seen = set(nums)
        target = original
        while target in seen:
            target <<= 1
        return target

File: src/rust/finished/keep_multiplying_found_values_by_two.rs
impl Solution {
    pub fn find_final_value(nums: Vec<i32>, original: i32) -> i32 {
        let seen: std::collections::HashSet<i32> = nums.into_iter().collect();
        let mut target = original;
        while seen.contains(&target) {
            target <<= 1;
        }
        target
    }
}
